pkg/store/cache/inmemory.go

```diff
@@ -302,6 +303,9 @@ func (c *InMemoryIndexCache) StorePostings(blockID ulid.ULID, l labels.Label, v
 // FetchMultiPostings fetches multiple postings - each identified by a label -
 // and returns a map containing cache hits, along with a list of missing keys.
 func (c *InMemoryIndexCache) FetchMultiPostings(_ context.Context, blockID ulid.ULID, keys []labels.Label) (hits map[labels.Label][]byte, misses []labels.Label) {
+	begin := time.Now()
+	defer c.commonMetrics.fetchLatency.WithLabelValues(cacheTypePostings).Observe(float64(time.Since(begin)))
```

Comment:
I think this will always be 0 because `time.Since` will get evaluated immediately and not when defer gets called. Might be worth creating a `prometheus.NewTimer` and then we can call `defer t.ObserveDuration()`.

Reply:
Thanks @fpetkovski! Good catch I have made the changes. PTAL